test/cri-containerd/container_test.go

```diff
@@ -476,3 +476,168 @@ func Test_RunContainer_ZeroVPMEM_Multiple_LCOW(t *testing.T) {
 	startContainer(t, client, ctx, containerIDTwo)
 	defer stopContainer(t, client, ctx, containerIDTwo)
 }
+
+func Test_RunContainer_GMSA_WCOW(t *testing.T) {
+	requireFeatures(t, featureWCOWProcess, featureGMSA)
+	csDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatalf("failed to create temp directory: %s", err)
+	}
+	defer func() {
+		if err := os.RemoveAll(csDir); err != nil {
+			t.Fatalf("failed deleting temp dir: %v", err)
+		}
+	}()
+	csPath := filepath.Join(csDir, testGmsaAccount+".json")
+	credSpec, err := gmsaSetup(csPath, testGmsaAccount)
+	if err != nil {
+		t.Fatalf("failed gmsa setup: %s", err)
+	}
+
+	pullRequiredImages(t, []string{imageWindowsNanoserver})
+	client := newTestRuntimeClient(t)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	sandboxRequest := &runtime.RunPodSandboxRequest{
+		Config: &runtime.PodSandboxConfig{
+			Metadata: &runtime.PodSandboxMetadata{
+				Name:      t.Name() + "-Sandbox",
+				Namespace: testNamespace,
+			},
+		},
+		RuntimeHandler: wcowProcessRuntimeHandler,
+	}
+
+	podID := runPodSandbox(t, client, ctx, sandboxRequest)
+	defer removePodSandbox(t, client, ctx, podID)
+	defer stopPodSandbox(t, client, ctx, podID)
+
+	request := &runtime.CreateContainerRequest{
+		PodSandboxId: podID,
+		Config: &runtime.ContainerConfig{
+			Metadata: &runtime.ContainerMetadata{
+				Name: t.Name() + "-Container",
+			},
+			Image: &runtime.ImageSpec{
+				Image: imageWindowsNanoserver,
+			},
+			Command: []string{
+				"cmd",
+				"/c",
+				"ping",
+				"-t",
+				"127.0.0.1",
+			},
+			Windows: &runtime.WindowsContainerConfig{
+				SecurityContext: &runtime.WindowsContainerSecurityContext{
+					CredentialSpec: credSpec,
+				},
+			},
+		},
+		SandboxConfig: sandboxRequest.Config,
+	}
+
+	containerID := createContainer(t, client, ctx, request)
+	defer removeContainer(t, client, ctx, containerID)
+	startContainer(t, client, ctx, containerID)
+	defer stopContainer(t, client, ctx, containerID)
+
+	// No klist and no powershell available
+	cmd := []string{"cmd", "/c", "set", "user"}
```

Comment:
You can just do `set USERDNSDOMAIN` instead.

Reply:
Wanted to make sure you didn't miss this.

Reply:
I feel just doing set user is better. Can show the format of the normal entries and what we expected to see along with them.

Reply:
Would you prefer I change it? I don't feel too strongly about it

Reply:
To me the way this is written makes it look like `set user` is some specific command, rather than just the `set` command with a pattern of `user`. It seems strange to not use a more specific command when we know we are looking for a specific variable here. If you feel you want to keep it this way, please at least add a comment explaining (that we are using the set command with a pattern of "user" to look for the USERDNSDOMAIN environment variable).

Reply:
No that's fair. I'll change it now